### ISSUE TITLE
C# dependencies no longer installs on Ubuntu 17.10

### ISSUE THREAD

**Issue by badersur** [dotnet/vscode-csharp#1903]
## Environment data
`dotnet --info` output:

```
.NET Command Line Tools (2.0.3-servicing-007131)

Product Information:
 Version:            2.0.3-servicing-007131
 Commit SHA-1 hash:  36e47bf417

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  17.10
 OS Platform: Linux
 RID:         linux-x64
 Base Path:   /home/badersur/Downloads/dotnet/sdk/2.0.3-servicing-007131/


Microsoft .NET Core Shared Framework Host

  Version  : 2.0.3
  Build    : a9190d4a75f4a982ae4b4fa8d1a24526566c69df
```

(installed using [my script](https://github.com/badersur/dotnet-manager))

VS Code version: 1.19.0-insider
C# Extension version: 1.14.0-beta2

## Steps to reproduce
Install latest stable version (or the beta)
Restart VSCode 

## Expected  behavior
The C# dependencies installs

## Actual behavior
I see this:

```
Installing C# dependencies...
Platform: linux, x86_64, name=ubuntu, version=17.10

Downloading package 'OmniSharp for Linux (x64)'
```

No matter how many hours I wait! I believe I had faced this issue before on Ubuntu 17.04 which I have solved by adding `"csharp.fallbackDebuggerLinuxRuntimeId": "ubuntu.16.10-x64"` to my settings but this setting doesn't seem to work now!

I'm using the latest alpha version of mono with the following packages installed:

git gconf2 gconf-service libgtk2.0-0 libudev1 libgcrypt20 libnotify4 libxtst6 libnss3 python gvfs-bin xdg-utils libcap2 libunwind-dev libicu-dev

Also, I'm using node 9 (and tried 8 and 7) installed using nvm (and I have added node & npm to the path by adding `export PATH=$PATH:/home/badersur/.nvm/versions/node/v9.2.0/bin` to */etc/profile.d/add-node.sh*)

Please help me to continue using your great extension :)

**Comment by badersur** [comment]
I have cloned the repo and used the fallback URL and the extension is working now!

I can't download: https://download.visualstudio.microsoft.com/download/pr/100289197/6b6deacae45d80bf0174c40b720ed20b/omnisharp-linux-x64-1.27.2.zip using chrome and firefox, and the connection timed out for no obvious reason...

By the way, it'd be awesome to document how to install the extension from source! I had to look at your travis file to find this out :cry:

**Comment by DustinCampbell** [comment]
@badersur : I'm glad to to hear you were able to get it working. From #1827, it sounds like you're having trouble downloading in general from Azure CDNs. Is that right?

**Comment by badersur** [comment]
I used to have issues with nuget as explained in https://github.com/NuGet/NuGetGallery/issues/3345, but that issue disappeared for a while without doing anything on my machine IIRC... 

In addition to that, I think I had issues with libraries such as https://ajax.aspnetcdn.com/ajax/bootstrap/3.3.6/css/bootstrap.min.css used on https://live.asp.net/ because I remember visiting the site and it was broken (which was temporal for few days/weeks?! not sure)

Strangely, I'm able to download https://www.microsoft.com/en-us/download/details.aspx?id=17851 but not https://download.visualstudio.microsoft.com/download/pr/100289197/6b6deacae45d80bf0174c40b720ed20b/omnisharp-linux-x64-1.27.2.zip!

**Comment by DustinCampbell** [comment]
Are you behind a proxy? I'm able to click on the link above and it downloads properly.

**Comment by badersur** [comment]
No, I'm connecting to my home router directly. I think the issue is related to either:

1. router's security settings (I remember setting it to the highest strict value, I forgot my password to double check!)
2. my ubuntu's ssl libraries are broken / miss configured (but other sites are working fine so I'm not really thinking so...
3. maybe have something to do with [HTTPS being set with SNI or IP](https://github.com/NuGet/NuGetGallery/issues/3345#issuecomment-260033642)

**Comment by badersur** [comment]
I'd like to add this note that may be useful:

I've upgraded to Ubuntu 17.10 from 17.04 but while installing the new packages I locked my screen and couldn't unlock it because the password box was unusable! I switched to using the terminal (Ctrl + Alt + F4) and tried to run `sudo apt-get update && sudo apt-get dist-upgrade -y` but running root commands is blocked because of the running installer. I've waited some hours until the upgrade process completed and reran the command. Then, everything worked fine!

**Comment by DustinCampbell** [comment]
> Then, everything worked fine!

Do you mean that the download is working for you now?

**Comment by badersur** [comment]
No, I mean Ubuntu has been upgraded without any noticeable issues after being stuck with a locked screen. This happened few months ago before facing the issue with C# extension in the last week.

I will try reinstalling my ubuntu packages as described in https://ubuntuforums.org/showthread.php?t=735693 and if it didn't help, I'll try to reformat my laptop.

**Comment by badersur** [comment]
Sorry for my late feedback.

I've tried to reinstall Ubuntu packages but every time I face an issue (power outage or network issues). Moreover, I've tried Ubuntu inside VirtualBox (without installing) and was able to download https://download.visualstudio.microsoft.com/download/pr/100289197/6b6deacae45d80bf0174c40b720ed20b/omnisharp-linux-x64-1.27.2.zip but I wasn't able to run VS Code!). Also, I couldn't visit plus.google.com inside VirtualBox...

I can't format my laptop because there's a bug in Ubuntu that [mess up with the BIOS](https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1734147).

Maybe my laptop is affected by some viruses..
